src/types.jl

```diff
@@ -235,6 +220,30 @@ cstruct(x::AcbRef) = x.acb_ptr
 Base.convert(::Type{Ptr{acb_struct}}, x::AcbRef) = cstruct(x)
 Base.cconvert(::Type{Ref{acb_struct}}, x::AcbRef) = cstruct(x)
 
+const MagLike = Union{Mag,cstructtype(Mag),Ptr{cstructtype(Mag)}}
+const ArfLike = Union{Arf,cstructtype(Arf),Ptr{cstructtype(Arf)}}
+const ArbLike = Union{Arb,ArbRef,cstructtype(Arb),Ptr{cstructtype(Arb)}}
+const AcbLike = Union{Acb,AcbRef,cstructtype(Acb),Ptr{cstructtype(Acb)}}
+const ArbVectorLike = Union{ArbVector,ArbRefVector,cstructtype(ArbVector)}
+const AcbVectorLike = Union{AcbVector,AcbRefVector,cstructtype(AcbVector)}
+const ArbMatrixLike = Union{ArbMatrix,ArbRefMatrix,cstructtype(ArbMatrix)}
+const AcbMatrixLike = Union{AcbMatrix,AcbRefMatrix,cstructtype(AcbMatrix)}
+const ArbTypes = Union{
+    Arf,
+    Arb,
+    ArbRef,
+    Acb,
+    AcbRef,
+    ArbVector,
+    AcbVector,
+    ArbMatrix,
+    AcbMatrix,
+    ArbRefVector,
+    AcbRefVector,
+    ArbRefMatrix,
+    AcbRefMatrix,
+}
```

Comment:
Minor thing, but can we either keep the `Ref` versions just under the default version or have all of them in the end, to make it easier to see directly what is going on.

Reply:
Done